backend/src/main/java/com/beaplab/BeaplabEngine/controller/WatchController.java

```diff
@@ -0,0 +1,336 @@
+package com.beaplab.BeaplabEngine.controller;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.json.simple.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.multipart.MultipartHttpServletRequest;
+import org.springframework.web.servlet.ModelAndView;
+
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import springfox.documentation.annotations.ApiIgnore;
+
+import com.beaplab.BeaplabEngine.authentication.SessionDetails;
+import com.beaplab.BeaplabEngine.constants.BeapEngineConstants;
+import com.beaplab.BeaplabEngine.service.AppleWatchService;
+import com.beaplab.BeaplabEngine.service.FitbitService;
+import com.rollbar.notifier.Rollbar;
+
+
+public class WatchController {
+
+    @Autowired
+    private final Rollbar rollbar;
+
+    /**
+     * injecting AppleWatchService into this class
+     */
+    @Autowired
+    private AppleWatchService appleWatchService;
+
+    /**
+     * injecting AppleWatchService into this class
+     */
+    @Autowired
+    private FitbitService fitbitService;
+
+    public WatchController(Rollbar rollbar){
+        this.rollbar = rollbar;
+    }
+
+    // want to handle api here. Get watch type from url parameter, call controller accordingly
+    
+
+    // getUploadView
+    @RequestMapping(value = "/{watchType}/uploadview", method = RequestMethod.GET)
+    @ApiOperation(value = "WatchType view", notes = "Returns the watchType view page", response = ModelAndView.class)
+        
+        public ModelAndView getUploadView(@PathVariable("watchType") String watchType) throws Exception{
+            rollbar.info("In WatchController/getUploadView with api parameter: " + watchType);
+            // getting the appropriate viewName
+            String viewName = "";
+            switch (watchType) {
+                case "applewatch":
+                    viewName = "appleWatch";
+                    break;
+                case "fitbit":
+                    viewName = "fitbit";
+                    break;
+                default:
+                    rollbar.error("Incorrect watch type selected: " + watchType);
+                    throw new Exception("Incorrect watch type selected: " + watchType);
```

Comment:
Is there a better way to show that an invalid watch type was passed in?

Reply:
i think this is fine